### PR TITLE
Split unit test service files up

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

- Use existing helper functions to remove duplication
- Split unit test service files into smaller chunks
- Driveby: Check existence of collections in db on family create/update and raise ValidationError in service layer if any don't exist

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [ ] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: tests/unit_tests/service/document/test_get_upload_details_document_service.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import app.service.document as doc_service
+
+
+def test_document_service_get_upload_details(test_s3_client):
+    result = doc_service.get_upload_details("path/file.ext", True)
```
Comment (diversemix) [inline tests/unit_tests/service/document/test_get_upload_details_document_service.py:5]: Forgot we even put this in!

File: app/repository/collection.py [hunk 2]
```diff
@@ -85,6 +85,18 @@ def _collection_to_dto(db: Session, co: CollectionOrg) -> CollectionReadDTO:
...
 
+def validate(db: Session, import_ids: set[str]) -> bool:
+    """Validate whether the given collection import IDs exist in the DB.
+
+    :param Session db: The DB session to connect to.
+    :param set[str] import_ids: The collection import IDs we want to
+        check the existence of.
+    :return bool: Return whether or not all the IDs exists in the DB.
+    """
+    collections = [col[0] for col in db.query(Collection.import_id).distinct().all()]
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply app/repository/collection.py:96]: Does that catch whether an import ID has been provided that does not exist in the database though?

File: tests/unit_tests/service/family/test_update_family_service.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+    updated_family = create_family_write_dto(collections=["x.y.z.2", "x.y.z.3"])
+
+    collection_repo_mock.invalid_org = True
+    with pytest.raises(ValidationError) as e:
+        family_service.update("a.b.c.d", USER_EMAIL, updated_family)
+    expected_msg = "Organisation mismatch between some collections and the current user"
+    assert e.value.message == expected_msg
+
+    assert family_repo_mock.get.call_count == 2
```
Comment (diversemix) [inline tests/unit_tests/service/family/test_update_family_service.py:291]: Begs the question if the assertion below actually matter - as long as the ValidationError was raised :thinking:

Comment (katybaulch) [reply tests/unit_tests/service/family/test_update_family_service.py:291]: Ah this was helpful for debugging the tests initially - there is some funky stuff going on with the mocks in some places. Would like to keep these in for now until we've re-written our mocks if poss

File: tests/unit_tests/service/document/test_get_upload_details_document_service.py (hunk 1 above)
Comment (katybaulch) [reply tests/unit_tests/service/document/test_get_upload_details_document_service.py:5]: Me too!

**Pull request merged**